http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java

```diff
@@ -397,7 +386,17 @@ public DefaultHttpClient(LoadBalancer loadBalancer, HttpClientConfiguration conf
         this(loadBalancer,
                 configuration, contextPath, new DefaultThreadFactory(MultithreadEventLoopGroup.class),
                 new NettyClientSslBuilder(new ClientSslConfiguration(), new ResourceResolver()),
-                createDefaultMediaTypeRegistry(), AnnotationMetadataResolver.DEFAULT);
+                createDefaultMediaTypeRegistry());
+    }
+
+    /**
+     * Factory method to create new instance of {@link DefaultHttpClient} with overridden {@link FilterResolver}.
+     *
+     * @param filterResolver filter resolver that new instance should be created with
+     * @return new instance of {@link DefaultHttpClient} with overridden {@link FilterResolver}.
+     */
+    public DefaultHttpClient withFilterResolver(FilterResolver filterResolver) {
```

Comment:
We don't have the builder pattern anywhere else in this file so I think this method is out of place

Reply:
I think it is a common pattern of how to deal with modification of immutable objects. You cannot change its state so you will need to create new instance with changed params. 

Taking into account `DefaultHttpClient` does not expose all its state through getters, it seems only class itself knows how it should be cloned. Do you want me to replace method name with ` DefaultHttpClient setFilterResolver()`? will it be less confusing?